test/unit/BondingManager.js

```diff
@@ -32,742 +47,1417 @@ contract("BondingManager", accounts => {
-            // Get Delegator 1 unbonded amount
-            const unbondedAmount = (await bondingManager.getDelegator(dAddr))[1]
-
-            const percPoints = Math.floor((delegatorStake * PERC_DIVISOR) / claimableStake)
-            const transcoderFeeShare = Math.floor((fees2 * (PERC_DIVISOR - feeShare)) / PERC_DIVISOR)
-            const delegatorsFeeShare = fees2 - transcoderFeeShare
-            const delegatorFeeShare = Math.floor((percPoints * delegatorsFeeShare) / PERC_DIVISOR)
-            const transcoderRewardShare = Math.floor((mintedTokens2 * blockRewardCut) / PERC_DIVISOR)
-            const delegatorsRewardShare = mintedTokens2 - transcoderRewardShare
-            const delegatorRewardShare = Math.floor((percPoints * delegatorsRewardShare) / PERC_DIVISOR)
-
-            const expDelegatorStake = add(delegatorStake, delegatorRewardShare).toString()
-            const expUnbondedAmount = add(unbondedAmount, delegatorFeeShare).toString()
-
-            await bondingManager.claimTokenPoolsShares(jobCreationRound2, {from: dAddr})
-
-            const dInfo = await bondingManager.getDelegator(dAddr)
-            assert.equal(dInfo[0].toString(), expDelegatorStake, "delegator stake incorrect")
-            assert.equal(dInfo[1].toString(), expUnbondedAmount, "delegator unbonded amount incorrect")
-
-            claimableStake -= delegatorStake
-            tokenPools = await bondingManager.getTranscoderTokenPoolsForRound(tAddr, jobCreationRound2)
-            assert.equal(tokenPools[0], mintedTokens2 - delegatorRewardShare, "wrong reward pool for token pools")
-            assert.equal(tokenPools[1], fees2 - delegatorFeeShare, "wrong fee pool for token pools")
-            assert.equal(tokenPools[3], claimableStake, "wrong claimable stake for token pools")
-
-            await bondingManager.claimTokenPoolsShares(jobCreationRound2, {from: tAddr})
-            tokenPools = await bondingManager.getTranscoderTokenPoolsForRound(tAddr, jobCreationRound2)
-            assert.equal(tokenPools[0], 0, "wrong reward pool for token pools")
-            assert.equal(tokenPools[1], 0, "wrong fee pool for token pools")
-            assert.equal(tokenPools[3], 0, "wrong claimable stake for token pools")
+        it("should fail if there are no fees to withdraw", async () => {
+            await expectThrow(bondingManager.withdrawFees({from: transcoder1}))
+        })
+
+        it("should withdraw caller's fees", async () => {
+            await bondingManager.claimEarnings(currentRound + 1, {from: transcoder0})
+            assert.isAbove((await bondingManager.getDelegator(transcoder0))[1], 0, "caller should have non-zero fees")
+
+            await bondingManager.withdrawFees({from: transcoder0})
+
+            const dInfo = await bondingManager.getDelegator(transcoder0)
+            assert.equal(dInfo[6], currentRound + 1, "should set caller's lastClaimRound")
+            assert.equal(dInfo[1], 0, "should set caller's fees to zero")
```

Comment:
Kind of along the lines of my comment about testing state changes specific to the contract under test - the actual ETH transfer is normally performed by a call to `trustedWithdrawETH` on the Minter, but since the Minter is mocked out in this unit test, that function call doesn't actually do anything. Similar to with the LivepeerToken contract, the idea was that actual ETH movement side effects that rely on the Minter could be tested in an integration test and this unit test would just focus on how the BondingManager state changes